zqd/packet/packet.go

```diff
@@ -74,64 +74,83 @@ func IngestFile(ctx context.Context, s *space.Space, pcap, zeekExec string) (*In
 		done:      make(chan struct{}),
 		zeekExec:  zeekExec,
 	}
+	if err = p.indexPcap(ctx); err != nil {
+		os.Remove(p.space.DataPath(IndexFile))
+		return nil, err
+	}
+	if err = p.space.SetPacketPath(p.pcapPath); err != nil {
+		os.Remove(p.space.DataPath(IndexFile))
+		return nil, err
+	}
 	go func() {
 		p.err = p.run(ctx)
 		close(p.done)
 	}()
-	return p, p.awaitZeekLogs()
+	return p, nil
 }
 
 func (p *IngestProcess) run(ctx context.Context) error {
-	idx, err := p.slurp(ctx)
-	if err != nil {
-		goto abort
+	var slurpErr error
+	slurpDone := make(chan struct{})
+	go func() {
+		slurpErr = p.slurp(ctx)
+		close(slurpDone)
+	}()
+
+	abort := func() {
+		os.RemoveAll(p.logdir)
+		os.Remove(p.space.DataPath(IndexFile))
+		p.space.SetPacketPath("")
 	}
-	if err = p.writeIndexFile(idx); err != nil {
-		goto abort
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	start := time.Now()
+	next := time.Second
+outer:
+	for {
+		select {
+		case <-slurpDone:
+			break outer
+		case t := <-ticker.C:
+			if t.After(start.Add(next)) {
+				if err := p.writeData(ctx); err != nil {
+					abort()
+					return err
+				}
+				next = 2 * next
+			}
+		}
 	}
-	if err = p.space.SetPacketPath(p.pcapPath); err != nil {
-		goto abort
+	if slurpErr != nil {
+		abort()
+		return slurpErr
 	}
-	if err = p.writeData(ctx); err != nil {
-		goto abort
+	if err := p.writeData(ctx); err != nil {
+		abort()
+		return err
 	}
-	if err = os.RemoveAll(p.logdir); err != nil {
-		goto abort
+	if err := os.RemoveAll(p.logdir); err != nil {
+		abort()
+		return err
 	}
 	return nil
-abort:
-	os.RemoveAll(p.logdir)
-	os.Remove(p.space.DataPath(IndexFile))
-	p.space.SetPacketPath("")
-	return err
 }
 
-// awaitZeekLogs waits for the first zeek logs to hit the file system. Should
-// an error occur before this happens, the error will be returned.
-func (p *IngestProcess) awaitZeekLogs() error {
-	w, err := fsnotify.NewWatcher()
+func (p *IngestProcess) indexPcap(ctx context.Context) error {
```

Comment:
```suggestion
func (p *IngestProcess) indexPcap() error {
```
ctx doesn't appear to be used here. This can be removed.

Reply:
I might be nice at some point to have `ctx` stop the copy, but I don't think this is important- the time it will take to create the index will be negligible compared to the rest of this procedure.

Reply:
Agreed - as I wrote offline `pcap index` is 💥 fast